gitea: 1.11.6 -> 1.12.0

Ma27 opened the pull request:
###### Motivation for this change

https://github.com/go-gitea/gitea/releases/tag/v1.12.0


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Ma27 committed:
gitea: 1.11.6 -> 1.12.0

https://github.com/go-gitea/gitea/releases/tag/v1.12.0

flokli commented:
https://github.com/go-gitea/gitea/releases/tag/v1.12.0 mentions "Security", and contains some code backported somewhere - so should we add a backport for stable as well?

FRidh commented:
"Security fix" is for "It is impossible to use -1 to disable algo for minimum key size check.".

flokli commented:
Oh wow, why would I want to disable the minimum key size check? oO.

kolaente commented:
> Oh wow, why would I want to disable the minimum key size check? oO.

@flokli You would not disable the key size check but the algorithm, see https://github.com/go-gitea/gitea/issues/11417#issuecomment-629565085 and https://github.com/go-gitea/gitea/blob/master/custom/conf/app.example.ini#L356-L361

So if you'd want to disable the use of say rsa keys, you would put an entry like this in the config: `RSA = -1`. The fix in the release fixes that because it was not working.

v1.11.7 has also been released, we should probably put that in stable instead. Though now with 1.12.0 release that's not getting updates for a long time.

Ma27 commented:
Just deployed this to my personal instance and it works fine. If everyone else agrees, I'd update gitea on stable to 1.11.7.

flokli commented:
@Ma27  SGTM.

aanderse reviewed (approved):
Thanks @Ma27! Enough approvals to merge :+1:

Ma27 commented:
Thanks! Will update gitea on release-20.03 to 1.11.8 tomorrow :)

Ma27 commented:
Oh and https://github.com/go-gitea/gitea/releases/tag/v1.12.1 is out as well now

Ma27 commented:
Updated on `stable` to [`1.11.8`](https://github.com/go-gitea/gitea/releases/tag/v1.11.8): 9c42eb70cfdb2e59c001a4cdae0b2c9875dd44c3
Updated on `master` to [`1.12.1`](https://github.com/go-gitea/gitea/releases/tag/v1.12.1): 14bc25038b1f2656bf30239ee7a04b0f863a7f3c
